### ISSUE TITLE
Improve warnings around unsafe types

### ISSUE THREAD

**Issue by ospencer** [grain-lang/grain#1153]
Right now, we check that unsafe types are not used with functions in Pervasives like `print`, but we should make the typechecker disallow them from typechecking with any generic type completely.

**Comment by ribeirotomas1904** [comment]
just to clarify
what this means, is that we should give a warning when an unsafe expression is used as an argument for where the function was expecting a parameter with generic type? (or would it be for a parameter that doesn't have type annotation?)

**Comment by ospencer** [comment]
Yeah, exactly.

**Comment by phated** [comment]
@ospencer The original body describes it happening in the type checker, not the "warnings" system. Can you clarify?

**Comment by ospencer** [comment]
Not quite sure what you mean 😅

**Comment by ribeirotomas1904** [comment]
by warning system, you mean what is being done [here](https://github.com/grain-lang/grain/blob/main/compiler/src/typed/typed_well_formedness.re#L101)?

**Comment by phated** [comment]
Yeah, I suppose I want clarification between "type checking" vs "wellformedness" which are different phases.

**Comment by ospencer** [comment]
Oh, we emit warnings in various phases. This would need to be done in the typechecking phase or typed well-formedness phase. I don't think we had typed well-formedness when this issue was originally created.

**Comment by ospencer** [comment]
(and typed well-formedness is different than the basic well-formedness)

**Comment by ospencer** [comment]
It's probably easiest to do in typed-wellformedness.
